extension/js/common/xss-safe-factory.ts

```diff
@@ -167,12 +176,15 @@ export class XssSafeFactory {
     return `<link class="${this.destroyableCls}" rel="stylesheet" href="${this.extUrl(`css/${file}.css`)}" />`;
   }
 
-  public dialogPassphrase = (longids: string[], type: PassphraseDialogType) => {
-    return this.divDialog_DANGEROUS(this.iframe(this.srcPassphraseDialog(longids, type), ['medium'], { scrolling: 'no' }), 'dialog-passphrase'); // xss-safe-factory
+  public showPassphraseDialog = async (longids: string[], type: PassphraseDialogType, initiatorFrameId?: string) => {
+    const result = await Ui.modal.iframe(this.srcPassphraseDialog(longids, type, initiatorFrameId), 500, 'dialog-passphrase');
+    if (result.dismiss === Swal.DismissReason.backdrop || result.dismiss === Swal.DismissReason.esc) {
```

Comment:
If I understand correctly, this piece of code relies on some knowledge of how `Ui.modal.iframe` is implemented, namely
that `DismissReason.cancel` or `DismissReason.timer` are not possible?
Why don't we use `if (result.dismiss !== DismissReason.close) ` or simply return boolean from `Ui.modal.iframe()`?

Reply:
> If I understand correctly, this piece of code relies on some knowledge of how `Ui.modal.iframe` is implemented, namely
> that `DismissReason.cancel` or `DismissReason.timer` are not possible?

Correct, we don't use timer or the SweetAlert2's cancel buttons in the passphrase dialog.

> Why don't we use `if (result.dismiss !== DismissReason.close) ` or simply return boolean from `Ui.modal.iframe()`?

Right, it can be as simple as checking `result.dismiss`. If a dialog was closed by `BrowserMsg.send.closeDialog()`, `result.dismiss` will be `undefined`. (docs: https://sweetalert2.github.io/#handling-dismissals)